lib/bolt/node/winrm.rb

```diff
@@ -297,8 +319,22 @@ def _run_task(task, input_method, arguments)
         Bolt::Node::Success.new
       end.then do
         with_remote_file(task) do |remote_path|
-          path, args = *process_from_extension(remote_path)
-          execute_process(path, args, stdin)
+          if is_powershell_file(remote_path) && stdin.nil?
+            execute(<<-PS)
+try
+{
+  # TODO: powershell doesn't support STDIN redirection when executing a script
```

Comment:
So we end up having to invoke `powershell.exe` anyhow if the task supports STDIN... maybe not a big deal because that combo should never be used in real life... but it makes the code more complex.